scripts/gulp/createBundle.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+var coffeeify = require('coffeeify');
+var browserify = require('browserify');
+var exorcist = require('exorcist');
+var fs = require('fs');
+var mkdirp = require('mkdirp');
+var uglifyify = require('uglifyify');
+var watchify = require('watchify');
+var gulpUtil = require('gulp-util');
+
+var log = gulpUtil.log;
+
+function streamFinished(stream) {
+  return new Promise(function (resolve, reject) {
+    stream.on('finish', resolve);
+    stream.on('error', reject);
+  });
+}
+
+function waitForever() {
+  return new Promise(function (resolve, reject) {});
+}
+
+/**
+ * type Transform = 'coffee';
+ *
+ * interface BundleOptions {
+ *   name: string;
+ *   path: string;
+ *
+ *   entry?: string[];
+ *   require?: string[];
+ *   transforms: Transform[];
+ *
+ *   watch?: boolean;
+ *   minify?: boolean;
+ * }
+ */
+
+/**
+ * Bundles the JavaScript for an application.
```

Comment:
TODO: Add docs explaining that this contains the shared logic for building a Browserify bundle with source maps and suitable transforms.